From 317fe23ed7a1834a4cff904cab570e14a5099f8a Mon Sep 17 00:00:00 2001
From: Ben Rudolph <brudolph@dimagi.com>
Date: Tue, 3 Jan 2017 11:23:45 +0200
Subject: [PATCH] locations for pull_users_and_groups

Fix limit_user_ids to limit both groups and locations

get user_ids at location and descendants

Move import

do not use import
---
 corehq/apps/reports/filters/users.py | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/corehq/apps/reports/filters/users.py b/corehq/apps/reports/filters/users.py
index 328bcd8..0b08aea 100644
--- a/corehq/apps/reports/filters/users.py
+++ b/corehq/apps/reports/filters/users.py
@@ -315,13 +315,17 @@ class ExpandedMobileWorkerFilter(BaseMultipleOptionFilter):
     @classmethod
     def pull_users_and_groups(cls, domain, mobile_user_and_group_slugs,
                               include_inactive=False, limit_user_ids=None):
-        user_ids = cls.selected_user_ids(mobile_user_and_group_slugs)
+        user_ids = set(cls.selected_user_ids(mobile_user_and_group_slugs))
         user_types = cls.selected_user_types(mobile_user_and_group_slugs)
         group_ids = cls.selected_group_ids(mobile_user_and_group_slugs)
+        location_ids = cls.selected_location_ids(mobile_user_and_group_slugs)
         users = []
 
+        if location_ids:
+            user_ids |= set(user_es.user_ids_at_locations_and_descendants(location_ids))
+
         if limit_user_ids:
-            user_ids = set(limit_user_ids).intersection(set(user_ids))
+            user_ids = set(limit_user_ids).intersection(user_ids)
 
         if user_ids or HQUserType.REGISTERED in user_types:
             users = util.get_all_users_by_domain(
-- 
2.2.1

